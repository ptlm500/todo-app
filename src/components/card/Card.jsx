import styled from 'styled-components';
import { theme } from '../../styles';

export default styled.div`
  background: ${theme.primaryLight};
  border-radius: ${theme.defaultRadius};
  padding: 1rem;
  width: fit-content;
`;
