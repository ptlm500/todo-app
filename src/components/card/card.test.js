import React from 'react';
import renderer from 'react-test-renderer';
import { Card, AnimatedCard } from './Card';
import 'jest-styled-components';

describe('Card', () => {
  it('renders a Card', () => {
    const tree = renderer.create(<Card />);
    expect(tree).toMatchSnapshot();
  });
});

describe('AnimatedCard', () => {
  it('renders an AnimatedCard', () => {
    const tree = renderer.create(
      <AnimatedCard animationClass={'test-class'} timeout={100}/>
    );
    expect(tree).toMatchSnapshot();
  });
});

