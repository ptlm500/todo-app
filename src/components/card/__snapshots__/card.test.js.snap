// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedCard renders an AnimatedCard 1`] = `
.c0 {
  background: #C5CAE9;
  border-radius: 0.5rem;
  padding: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0.card-transition-enter {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.card-transition-enter-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.card-transition-appear {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.card-transition-appear-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.test-class-enter {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.test-class-enter-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 100ms ease-out;
  transition: 100ms ease-out;
}

.c0.test-class-appear {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.test-class-appear-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 100ms ease-out;
  transition: 100ms ease-out;
}

<div
  className="sc-AxjAm c0"
/>
`;

exports[`Card renders a Card 1`] = `
.c0 {
  background: #C5CAE9;
  border-radius: 0.5rem;
  padding: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0.card-transition-enter {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.card-transition-enter-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0.card-transition-appear {
  opacity: 0.01;
  -webkit-transform: scaleX(0) scaleY(0);
  -ms-transform: scaleX(0) scaleY(0);
  transform: scaleX(0) scaleY(0);
}

.c0.card-transition-appear-active {
  opacity: 1;
  -webkit-transform: scaleX(1) scaleY(1);
  -ms-transform: scaleX(1) scaleY(1);
  transform: scaleX(1) scaleY(1);
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

<div
  className="c0"
/>
`;
